### PR TITLE
Completes UsersServlet.java code for review.

### PR DESCRIPTION
**Pull request by Cayle**

### CODE REVIEW

**Comment by Cayle** [comment]
> close all branches that dont require review
> consider only having essentials in doPost so that it reads easier .
> I'd also suggest rearranging so that the doPost and doGet are at the top so its easier to trace which helper functions are being used when to do what

Done .

File: capstone/src/main/java/com/google/sps/servlets/UsersServlet.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    Boolean isUserLoggedIn = userService.isUserLoggedIn();
+    Boolean IsUserRegistered;
+    if (!isUserLoggedIn) {
+      response.sendRedirect("/login");
+      return;
+    }
+    try {
+      IsUserRegistered = IsUserRegistered(pool, userService, response, request);
+      if (!IsUserRegistered) {
+        response.setStatus(300);
```
Comment (ljfranklin) [inline capstone/src/main/java/com/google/sps/servlets/UsersServlet.java:84]: I've never seen anyone use a plain 300 error code, usually you just see `301` or `307` for redirects. Probably could be simplified even more but at this point let's just leave this as is.

File: capstone/src/main/java/com/google/sps/servlets/UsersServlet.java [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    response.setContentType("application/json");
+    ServletContext servletContext = getServletContext();
+    DataSource pool = (DataSource) servletContext.getAttribute("my-pool");
+    if (!request.getPathInfo().equals("/current")) {
+      response.sendError(404, "Invalid url supplied by the client.");
+      return;
+    }
+    UserService userService = UserServiceFactory.getUserService();
+    Boolean isUserLoggedIn = userService.isUserLoggedIn();
+    Boolean IsUserRegistered;
```
Comment (ljfranklin) [inline capstone/src/main/java/com/google/sps/servlets/UsersServlet.java:76]: First letter of variable should be lowercase.

File: capstone/src/main/java/com/google/sps/servlets/UsersServlet.java [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+   public Boolean isPostRequestAuthorized(UserService userService, DataSource pool, HttpServletResponse response) 
+   throws IOException{
+       Boolean isUserLoggedIn = userService.isUserLoggedIn();
+        if (!isUserLoggedIn) {
+            response.sendRedirect("/login");
+            return false;
+        }
+        String email = userService.getCurrentUser().getEmail();
+        String logoutUrl = userService.createLogoutURL(URL_TO_REDIRECT_AFTER_USER_LOGS_OUT);
+        if (!email.endsWith("google.com")) {
```
Comment (ljfranklin) [inline capstone/src/main/java/com/google/sps/servlets/UsersServlet.java:288]: Be very careful with anything security related. Double, triple check it for any edge cases. Ask someone for help if you're unsure. I could go register totally-not-google.com and my email would get past this check. I _think_ `endsWith("@google.com")` is safe but for production code I would ask a security expert. Email address can have all sorts of weird characters: https://en.wikipedia.org/wiki/Email_address#Syntax.

File: capstone/src/main/java/com/google/sps/servlets/UsersServlet.java (hunk 1 above)
Comment (Cayle) [reply capstone/src/main/java/com/google/sps/servlets/UsersServlet.java:84]: Okay

**Pull request merged**